### PR TITLE
add master merge to release branch

### PR DESCRIPTION
**Pull request by alvin-huang**
This PR will add a nightly auto merge of `master` into `release/1-6` at `2AM UTC <> 10PM EST <> 7PM PST`. 

I decided to keep the code in here rather than in a bash script since it will most likely go away after the release is done. The high level steps of the configuration are:
* create a branch called `ci/master-merge-$(date +%Y%m%d%H%M%S)`. The latest `master` commit will be merged in here
* The new branch will be pushed up to GitHub and will trigger a CircleCI run
* Poll the CircleCI branch jobs until they are all complete and see if any fail
  * If jobs fail, display them and delete the merge branch
  * If jobs pass, checkout the release branch, merge master, and push it back up to the release branch. Then delete the temporary merge branch.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -429,8 +429,85 @@ jobs:
...
+            git config --global user.name "hashicorp-ci"
+
+            # Fetch latest master
+            git fetch origin
+
+            # Create a merge branch to run tests on
+            git_merge_branch="ci/master-merge-$(date +%Y%m%d%H%M%S)"
+            git checkout -b "${git_merge_branch}"
+            latest_oss_commit="$(git rev-parse origin/master)"
+            git merge -m "Merge Consul OSS branch 'master' at commit ${latest_oss_commit}" origin/master
```
Comment (i0rek) [inline .circleci/config.yml]:
```suggestion
            git merge -m "Merge Consul OSS branch 'master' at commit ${latest_oss_commit}" "${latest_oss_commit}"
```

That way you don't merge commits that happen to happen in between this and the previous statement. Very hypothetical.

File: .circleci/config.yml [hunk 2]
```diff
@@ -429,8 +429,85 @@ jobs:
...
+            echo "Waiting for master merge branch CI pipeline to finish..."
+            builds=
+            unfinished_builds="will be populated in the until loop below"
+            min_waited=0
+            until [[ -z "${unfinished_builds}" ]]; do
+              builds="$(curl \
+                --header 'Accept: application/json' \
+                --show-error \
+                --silent \
+                --user "${CIRCLECI_API_TOKEN}:" \
```
Comment (i0rek) [inline .circleci/config.yml]: Is the trailing colon correct?

Comment (alvin-huang) [reply .circleci/config.yml]: That's probably a typo but actually I don't need a token for a public repo so I'll just remove this line.

File: .circleci/config.yml (hunk 1 above)
Comment (alvin-huang) [reply .circleci/config.yml]: resolved in the following commit

**Pull request merged**